Vic2ToHoI4/Source/V2World/Mods/ModFactory.cpp

```diff
@@ -0,0 +1,24 @@
+#include "ModFactory.h"
+#include "ParserHelpers.h"
+
+
+
+Vic2::Mod::Factory::Factory()
+{
+	registerKeyword("name", [this](const std::string& unused, std::istream& theStream) {
+		mod->name = commonItems::singleString{theStream}.getString();
+	});
+	registerKeyword("path", [this](const std::string& unused, std::istream& theStream) {
+		mod->directory = commonItems::singleString{theStream}.getString();
+		mod->directory = mod->directory.substr(4, mod->directory.size());
+	});
+	registerRegex(commonItems::catchallRegex, commonItems::ignoreItem);
+}
+
+
+std::unique_ptr<Vic2::Mod> Vic2::Mod::Factory::getMod(const std::string& modFile, const std::string& Vic2ModDir)
+{
+	mod = std::make_unique<Mod>();
```

Comment:
Check both name and path exist and are valid before pushing. People do silly stuff.

Reply:
Done